packages/editor/src/components/post-schedule/index.js

```diff
@@ -17,10 +17,18 @@ function getDayOfTheMonth( date = new Date(), firstDay = true ) {
 	return new Date(
 		d.getFullYear(),
 		d.getMonth() + ( firstDay ? 0 : 1 ),
-		firstDay ? 1 : 0
+		firstDay ? 1 : 0,
+		firstDay ? 0 : 23,
+		firstDay ? 0 : 59,
+		firstDay ? 0 : 59
 	).toISOString();
 }
```

Comment:
Let's add `date-fns` `^2.28.0` as a dependency of `@wordpress/editor` (`packages/editor/package.json`) and use `startOfMonth` and `endOfMonth`. This will mean less code in the generated bundle sent to users because `@wordpress/components` already uses these functions.

Reply:
Thanks for sharing that dependency. I've switched over to that in 71b97c4eb5

Reply:
I spoke to soon. It looks like this commit caused testing to fail (confirmed by [reverting that commit on a test branch](https://github.com/WordPress/gutenberg/pull/45047)).

I'll get that fixed then give you a ping when it's ready for another look.

Reply:
@noisysocks Based on [the output of the tests](https://github.com/WordPress/gutenberg/pull/44540/commits/f06f64c0970290e14279390ce246752a6315cdca#diff-3836a6db831ccfe3af87b4c80a7fa67c44eb8d4239cff2dbfe3025924cd4e991), I think `date-fns` is causing problems with other functions within `@wordpress/editor` because it overloads some functions and expects ISO date strings:

> Starting with v2.0.0-beta.1 date-fns doesn't accept strings as date arguments. Please use `parseISO` to parse strings. See: https://github.com/date-fns/date-fns/blob/master/docs/upgradeGuide.md#string-arguments

Based on this, adding `date-fns` may be out of scope for this PR and may need some more work. Should I revert this change or do we need to tackle this here?

Reply:
Hi! Sorry for the delay, I was away for a week (support rotation).

Hop you don't mind I pushed up a change which should straighten this out a little and fix the errors. This stuff gets confusing because there are subtle differences between the different date representations. `postDate` is an ISO-like (but not ISO) string, `DateTimePicker` accepts anything, `onMonthPreviewed` gives back an ISO string, and `date-fns` functions want only `Date` objects 😅

Reply:
Thanks for this @noisysocks !

> Hope you don't mind I pushed up a change which should straighten this out a little and fix the errors. 
Not at all! I appreciate the help on this one.

> This stuff gets confusing because there are subtle differences between the different date representations. postDate is an ISO-like (but not ISO) string, DateTimePicker accepts anything, onMonthPreviewed gives back an ISO string, and date-fns functions want only Date objects 😅

Wow, that's a lot to unpack! I'm grateful you got this updated to work right. Now I just need to unravel it to grasp it all for my future self 🤣

Reply:
It's a mess lol. Maybe one day we can improve the API so that it's all `Date` objects. Hard to do without affecting backwards compatibility though.